src/v/cluster/topic_table.cc

```diff
@@ -681,6 +684,70 @@ topic_table::apply(set_topic_partitions_disabled_cmd cmd, model::offset o) {
     co_return errc::success;
 }
 
+ss::future<std::error_code>
+topic_table::apply(bulk_force_reconfiguration_cmd cmd, model::offset o) {
+    _last_applied_revision_id = model::revision_id(o);
+    auto validation_ec = validate_force_reconfigurable_partitions(
+      cmd.value.user_approved_force_recovery_partitions);
+    if (validation_ec) {
+        co_return validation_ec;
+    }
+    for (const auto& entry :
+         cmd.value.user_approved_force_recovery_partitions) {
+        add_partition_to_force_reconfigure(entry);
+    }
+    co_return errc::success;
+}
+
+std::error_code topic_table::validate_force_reconfigurable_partitions(
+  const fragmented_vector<ntp_with_majority_loss>& partitions) const {
+    std::error_code last_ec = errc::success;
+    for (const auto& entry : partitions) {
+        const auto& ntp = entry.ntp;
+        const auto& topic_md = get_topic_metadata_ref({ntp.ns, ntp.tp.topic});
+        vlog(
+          clusterlog.debug,
+          "Processing force reconfiguration entry {}, previous entry result: "
+          "{}",
+          entry,
+          last_ec);
+        if (entry.defunct_nodes.size() <= 0) {
+            last_ec = errc::invalid_request;
```

Comment:
should we simply return first error code ?

Reply:
I did this for logging (so we log all errors in the command, than just one).

Reply:
this may be a little bit confusing to log current entry and previous error, maybe if we would extract a function executing validation of a single entry we could do something like this: 
```
for (const auto& entry : partitions) {
   auto ec = validate_entry(entry)
   log.debug(...)
 }
```
WDYT ?

Reply:
Good idea done.